src/SpeedyTransforms/fourier.jl

```diff
@@ -0,0 +1,41 @@
+"""$(TYPEDSIGNATURES)
+Inverse fast Fourier transform of Legendre-transformed inputs `g_north` and `g_south` to be stored in `grids`.
+Not to be called directly, use `transform!` instead."""
+function _fourier!(
+    grids::AbstractGridArray,               # gridded output
```

Comment:
Do we want to do this as a dispatch based on the array type, or do we want to add the device type to the `SpectralTransform` struct to make the dispatch based on that? 

For me, the second option sounds better, I think.

Reply:
I agree, the `SpectralTransform` should decide where to compute things, and then we can probably even have convenience functions around it so that a CPU array is moved to the GPU e.g.